config/7.3.0/securebanking/ig/scripts/groovy/ObResponseCheck.groovy

```diff
@@ -99,6 +99,21 @@ static isObCompliantError(responseBody) {
   return false
 }
 
+boolean isV3Request(def request) {
+  String apiVersionRegex = "(v\\d+)"
+  def match = (request.uri.path =~ apiVersionRegex)
+  def isV3Request = true
+  if (match.find()) {
+    def apiVersion = match.group(1)  // Capture only the major version, e.g., "v4"
+    if (apiVersion == "v4") {
+      return false
+    }
+  }
+  return true;
```

Comment:
Can you rewrite this not to use a regex please?

The groovy `=~` operator is basically syntactic sugar for `java.util.regex.Pattern.compile(string)` which has a relatively high performance cost, and one compile call is being made each time is this method is called.

As the path element containing the version is constant, then its more efficient to manually parse the String. The `org.forgerock.http.MutableUri#getPathElements` method of request.uri field is handy to get the path elements as an array.